tools/FileTestDefs.sh

```diff
@@ -666,6 +666,33 @@ check_throws_test() {
 }
 standard_checks+=(check_throws)
 
+# Check for typos similar to Doxygen groups
+check_dox_groups() {
+    is_c++ "$1" && staged_grep -E "^\s*//+\s*[\{\}\(\)]*@[\{\}\(\)]*" "$1" \
+            | grep -v -E "(^\s*///@\{|^\s*///@\})"
+}
+check_dox_groups_report() {
+    echo "Found a likely typo similar to a valid doxygen grouping"
+    echo "Use balanced ///@{ and ///@} (on their own lines) to"\
+         " create doxygen groups"
+    pretty_grep -E "^\s*//+\s*[\{\}\(\)]*@[\{\}\(\)]*" "$1" | \
+        grep -v -E "(///@{|///@})"
+}
+check_dox_groups_test() {
+    test_check pass foo.hpp '///@{'
+    test_check pass foo.hpp '///@}'
+    test_check pass foo.txt '// @('
```

Comment:
I don't think you use this `// @(` syntax anywhere, is it valid / did you intend to prohibit it?

Reply:
This test checks that we don't error when the offending string appears in non-cpp files -- note `foo.txt` instead of `foo.hpp` (this same check is done for many of the git hook checks in this file)